### PR TITLE
session storage now actually can store the sessions without panics

### LINKED ISSUE #814: Service crashes after 6th client connected (panic: runtime error: index out of range)
```
C:\Program Files\MysteriumVPN\bin> .\myst.exe --testnet service openvpn --agreed-terms-and-conditions
2019-03-21T15:45:46.0825054 [Info] Starting Mysterium Node (0.0.0-dev)
2019-03-21T15:45:46.1005195 [Info] [Openvpn check] OpenVPN 2.4.6 x86_64-w64-mingw32 [SSL (OpenSSL)] [LZO] [LZ4] [PKCS11] [AEAD] built on May 15 2018
2019-03-21T15:45:46.1005195 [Info] [Openvpn check] library versions: OpenSSL 1.1.0h  27 Mar 2018, LZO 2.10
2019-03-21T15:45:46.1005195 [Info] [Openvpn check] Windows version 6.2 (Windows 8 or greater) 64bit
2019-03-21T15:45:46.1005195 [Info] [Openvpn check] Originally developed by James Yonan
2019-03-21T15:45:46.1005195 [Info] [Openvpn check] Copyright (C) 2002-2018 OpenVPN Inc <sales@openvpn.net>
2019-03-21T15:45:46.1005195 [Info] [Openvpn check] Custom tag: Compile time defines: enable_async_push=no enable_comp_stub=no enable_crypto=yes enable_crypto_ofb_cfb=yes enable_debug=no enable_def_auth=yes enable_dependency_tracking=no enable_dlopen=unknown enable_dlopen_self=unknown enable_dlopen_self_static=unknown enable_fast_install=needless enable_fragment=yes enable_iproute2=no enable_libtool_lock=yes enable_lz4=yes enable_lzo=yes enable_management=yes enable_multihome=yes enable_pam_dlopen=no enable_pedantic=no enable_pf=yes enable_pkcs11=yes enable_plugin_auth_pam=no enable_plugin_down_root=no enable_plugins=no enable_port_share=yes enable_selinux=no enable_server=yes enable_shared=no enable_shared_with_static_runtimes=yes enable_silent_rules=no enable_small=no enable_static=yes enable_strict=no enable_strict_options=no enable_systemd=no enable_werror=no enable_win32_dll=yes enable_x509_alt_username=no with_aix_soname=aix with_crypto_library=openssl with_gnu_ld=yes with_mem_check=no with_special_build=standalone-2.4.6-2-62 with_sysroot=no
...
...


panic: runtime error: index out of range

goroutine 148 [running]:
github.com/mysteriumnetwork/node/session.(*StorageMemory).Find(0xc0002ad200, 0xc00001afd5, 0x24, 0x0, 0x0, 0x0, 0x0, 0x0, 0xc000ad3200)
        /ext-go/1/src/github.com/mysteriumnetwork/node/session/storage_memory.go:65 +0x172
github.com/mysteriumnetwork/node/services/openvpn/session.(*clientMap).FindClientSession(0xc000b88040, 0x6, 0xc00001afd5, 0x24, 0x0, 0x0, 0x0, 0x0, 0x0, 0x5, ...)
        /ext-go/1/src/github.com/mysteriumnetwork/node/services/openvpn/session/client_map.go:44 +0x9e
github.com/mysteriumnetwork/node/services/openvpn/session.(*Validator).Validate(0xc000b88020, 0x6, 0xc00001afd5, 0x24, 0xc0002c6a95, 0x58, 0x3, 0xf72860, 0xc000b2f8c0)
        /ext-go/1/src/github.com/mysteriumnetwork/node/services/openvpn/session/validator.go:52 +0x90
github.com/mysteriumnetwork/node/services/openvpn/session.(*Validator).Validate-fm(0x6, 0xc00001afd5, 0x24, 0xc0002c6a95, 0x58, 0x90000000064f298, 0x40d479, 0x0)
        /ext-go/1/src/github.com/mysteriumnetwork/node/services/openvpn/service/factory.go:83 +0x63
github.com/mysteriumnetwork/node/vendor/github.com/mysteriumnetwork/go-openvpn/openvpn/middlewares/server/auth.(*middleware).authenticateClient(0xc000ab9590, 0x6, 0x2, 0xc00001afd5, 0x24, 0xc0002c6a95, 0x58, 0x6, 0x0)
        /ext-go/1/src/github.com/mysteriumnetwork/node/vendor/github.com/mysteriumnetwork/go-openvpn/openvpn/middlewares/server/auth/middleware.go:177 +0x265
github.com/mysteriumnetwork/node/vendor/github.com/mysteriumnetwork/go-openvpn/openvpn/middlewares/server/auth.(*middleware).handleClientEvent(0xc000ab9590, 0xc00001ce28, 0x6, 0x6, 0x2, 0xc000142780)
        /ext-go/1/src/github.com/mysteriumnetwork/node/vendor/github.com/mysteriumnetwork/go-openvpn/openvpn/middlewares/server/auth/middleware.go:156 +0x141
github.com/mysteriumnetwork/node/vendor/github.com/mysteriumnetwork/go-openvpn/openvpn/middlewares/server/auth.(*middleware).endOfEvent(0xc000ab9590)
        /ext-go/1/src/github.com/mysteriumnetwork/node/vendor/github.com/mysteriumnetwork/go-openvpn/openvpn/middlewares/server/auth/middleware.go:143 +0x51
github.com/mysteriumnetwork/node/vendor/github.com/mysteriumnetwork/go-openvpn/openvpn/middlewares/server/auth.(*middleware).ConsumeLine(0xc000ab9590, 0xc000ade020, 0xf, 0xc000b731b0, 0x0, 0x0)
        /ext-go/1/src/github.com/mysteriumnetwork/node/vendor/github.com/mysteriumnetwork/go-openvpn/openvpn/middlewares/server/auth/middleware.go:108 +0x5a8
github.com/mysteriumnetwork/node/vendor/github.com/mysteriumnetwork/go-openvpn/openvpn/management.(*Management).deliverOpenvpnManagementEvents(0xc0001bebd0, 0xc000052420)
        /ext-go/1/src/github.com/mysteriumnetwork/node/vendor/github.com/mysteriumnetwork/go-openvpn/openvpn/management/listener.go:226 +0x323
github.com/mysteriumnetwork/node/vendor/github.com/mysteriumnetwork/go-openvpn/openvpn/management.(*Management).serveNewConnection.func2(0xc00014e010, 0xc0001bebd0, 0xc000052420)
        /ext-go/1/src/github.com/mysteriumnetwork/node/vendor/github.com/mysteriumnetwork/go-openvpn/openvpn/management/listener.go:167 +0x62
created by github.com/mysteriumnetwork/node/vendor/github.com/mysteriumnetwork/go-openvpn/openvpn/management.(*Management).serveNewConnection
        /ext-go/1/src/github.com/mysteriumnetwork/node/vendor/github.com/mysteriumnetwork/go-openvpn/openvpn/management/listener.go:165 +0x2ea
```

### PR DESCRIPTION
**Pull request by vkuznecovas**
closes #814

### CODE REVIEW

File: session/storage_memory.go [hunk 1]
```diff
@@ -18,51 +18,57 @@
...
+	// see Benchmark_Storage_GetAll
+	sessions := make([]Session, len(storage.sessions))
+
+	i := 0
+	for _, value := range storage.sessions {
+		sessions[i] = value
+		i++
+	}
+
+	sort.Slice(sessions, func(i, j int) bool { return sessions[i].CreatedAt.Before(sessions[j].CreatedAt) })
```
Comment (donce) [inline session/storage_memory.go]: Do we need it to be sorted by createdAt? If we need this ordering only in endpoint, maybe we can perform sorting in there?

File: session/storage_memory.go [hunk 2]
```diff
@@ -77,8 +83,7 @@ func (storage *StorageMemory) Remove(id ID) {
 	storage.lock.Lock()
 	defer storage.lock.Unlock()
 
-	if sessionIndex, found := storage.idToIndex[id]; found {
-		delete(storage.idToIndex, id)
-		storage.sessions = append(storage.sessions[:sessionIndex], storage.sessions[sessionIndex+1:]...)
+	if _, found := storage.sessions[id]; found {
+		delete(storage.sessions, id)
```
Comment (donce) [inline session/storage_memory.go]: We don't need `if` check:
> If the requested key doesn't exist, we get the value type's zero value.

File: session/storage_memory_test.go [hunk 3]
```diff
@@ -56,51 +58,71 @@ func TestStorage_Add(t *testing.T) {
...
-		},
-		storage.idToIndex,
-	)
 }
 
 func TestStorage_GetAll(t *testing.T) {
-	sessionFirst := Session{ID: ID("id1")}
-	sessionSecond := Session{ID: ID("id2")}
+	sessionFirst := Session{
+		ID: ID("id1"),
```
Comment (donce) [inline session/storage_memory_test.go]: No `CreatedAt`?

File: session/storage_memory.go (hunk 2 above)
Comment (vkuznecovas) [reply session/storage_memory.go]: good point

File: session/storage_memory_test.go (hunk 3 above)
Comment (vkuznecovas) [reply session/storage_memory_test.go]: defaults to an old date - intended.

File: tequilapi/endpoints/service_sessions.go [hunk 4]
```diff
@@ -70,11 +70,7 @@ func NewServiceSessionsEndpoint(sessionStorage serviceSessionStorage) *serviceSe
 //     schema:
 //       "$ref": "#/definitions/ErrorMessageDTO"
 func (endpoint *serviceSessionsEndpoint) List(resp http.ResponseWriter, request *http.Request, params httprouter.Params) {
-	sessions, err := endpoint.sessionStorage.GetAll()
-	if err != nil {
-		utils.SendError(resp, err, http.StatusInternalServerError)
```
Comment (tadaskay) [inline tequilapi/endpoints/service_sessions.go]: Now that this is removed, is the swagger metadata still relevant?

```
500:
//     description: Internal server error
//     schema:
//       "$ref": "#/definitions/ErrorMessageDTO"
```

File: session/storage_memory_test.go [hunk 5]
```diff
@@ -91,6 +91,14 @@ func TestStorage_Remove(t *testing.T) {
 	assert.Len(t, storage.sessions, 0)
 }
 
+func TestStorage_RemoveLast(t *testing.T) {
```
Comment (tadaskay) [inline session/storage_memory_test.go]: As far as I understand, this tries to remove a non-existent session ID. How is it different from `TestStorage_Remove_Does_Not_Panic`?

Comment (vkuznecovas) [reply session/storage_memory_test.go]: TestStorage_Remove_Does_Not_Panic performs a few deletes on existing IDs whereas this one actually tries to remove a non existing one.

File: tequilapi/endpoints/service_sessions.go (hunk 4 above)
Comment (vkuznecovas) [reply tequilapi/endpoints/service_sessions.go]: Good catch

File: session/storage_memory_test.go [hunk 6]
```diff
@@ -56,51 +58,80 @@ func TestStorage_Add(t *testing.T) {
...
 
 func TestStorage_Remove(t *testing.T) {
 	storage := mockStorage(sessionExisting)
 
 	storage.Remove(sessionExisting.ID)
 	assert.Len(t, storage.sessions, 0)
-	assert.Len(t, storage.idToIndex, 0)
+}
+
+func TestStorage_RemoveLast(t *testing.T) {
```
Comment (donce) [inline session/storage_memory_test.go]: Hm, why "last"?
```suggestion
func TestStorage_RemoveNonExisting(t *testing.T) {
```

File: session/storage_memory_test.go (hunk 5 above)
Comment (tadaskay) [reply session/storage_memory_test.go]: Thanks

**Pull request merged**